src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs

```diff
@@ -0,0 +1,218 @@
+using System;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Serilog;
+
+namespace Datadog.Trace.Ci
+{
+    internal static class CIEnvironmentValues
+    {
+        private static readonly Regex BranchRegex = new Regex(@"^refs\/heads\/(.*)|refs\/(.*)$", RegexOptions.Compiled);
+        private static readonly ILogger Logger = DatadogLogging.GetLogger(typeof(CIEnvironmentValues));
+
+        static CIEnvironmentValues()
+        {
+            if (EnvironmentHelpers.GetEnvironmentVariable("TRAVIS") != null)
+            {
+                IsCI = true;
+                Provider = "travis";
+                Repository ??= EnvironmentHelpers.GetEnvironmentVariable("TRAVIS_REPO_SLUG");
+                Commit ??= EnvironmentHelpers.GetEnvironmentVariable("TRAVIS_COMMIT");
+                SourceRoot ??= EnvironmentHelpers.GetEnvironmentVariable("TRAVIS_BUILD_DIR");
+                PipelineId = EnvironmentHelpers.GetEnvironmentVariable("TRAVIS_BUILD_ID");
```

Comment:
Some of these use `??=` and others `=`. What this intentional? Seems to me like they would _all_ be null when this static ctor runs.

Reply:
You're right, good catch!, copy/pasting problem... I'll change that.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/802/commits/2b2ee99680880c3f41467916e08fecbb2deb93dd